scalafix-core/src/main/scala/scalafix/internal/patch/ReplaceSymbolOps.scala

```diff
@@ -126,10 +143,20 @@ object ReplaceSymbolOps {
             if sig.name != parent.value =>
           Patch.empty // do nothing because it was a renamed symbol
         case Some(parent) =>
+          val causesCollision = getGlobalImports(ctx.tree).exists { importStmt =>
+            importStmt.importers.flatMap(_.importees).exists {
+              case Importee.Name(name) => name.value == to.signature.name
+              case Importee.Rename(_, rename) => rename.value == to.signature.name
+              case _ => false
```

Comment:
Wildcards can also cause collisions (although we can't tell using Scalafix APIs). This is probably outside the scope of this PR, but we could have an opt-in "safe" flag that disable global imports whenever it finds wildcard imports.